### PR TITLE
[Core,FluidDyn] fix warnings in windows

### PR DESCRIPTION
**Pull request by ddiezrod**
**Description**
Removed a couple of warnings in Visual Studio.

**Changelog**
-Move MASS_SOURCE to variables (it was repeated in FluidDynApp and dem_variables.h)
-Discard returned value in global_pointer_utilities.h. This removes the warning, but I think there is an error there. std::unique does not remove the repeated elements unless you do it explicitly. I understand it should be:

```
auto last = std::unique(all_remote_ids.begin(), all_remote_ids.end());
all_remote_ids.erase(last, all_remote_ids.end());
```

EDIT:
The final result outcome of this PR is:
-MASS_SOURCE was removed from core, so now its only used in FluidDynamics.
-WX flag was added in windows compilation, so all warnings are turned into errors. To achieve this, some small corrections were needed, including the removal of the deprecated constructor in the LinearStrategy.
-Corrected some std::unique in global_pointer_utilities and in colouring_communication_utilities where duplicated entities were not being removed from their respective vectors.

### CODE REVIEW

File: kratos/utilities/global_pointer_utilities.h [hunk 1]
```diff
@@ -180,7 +180,7 @@ class GlobalPointerUtilities
         if(current_rank == master_rank)
         {
             std::sort(all_remote_ids.begin(), all_remote_ids.end());
-            std::unique(all_remote_ids.begin(), all_remote_ids.end());
+            static_cast<void>(std::unique(all_remote_ids.begin(), all_remote_ids.end()));
```
Comment (philbucher) [inline kratos/utilities/global_pointer_utilities.h]: yikes :sweat_smile: 
I think you should rather fix it properly, I am pretty sure whoever implemented `[[nodiscard]]` here had their reason

@RiccardoRossi can you have a look please?

Comment (ddiezrod) [reply kratos/utilities/global_pointer_utilities.h]: Yep, just to remark that my proposal is actually the one above:

```
auto last = std::unique(all_remote_ids.begin(), all_remote_ids.end());
all_remote_ids.erase(last, all_remote_ids.end());
```

**Comment by philbucher** [comment]
in order to keep the compilation clean in the future, could you please add the warnings [here](https://github.com/KratosMultiphysics/Kratos/blob/master/.github/workflows/configure.cmd#L40) such that they will be turned into errors in the CI? In this PR

**Comment by ddiezrod** [comment]
> in order to keep the compilation clean in the future, could you please add the warnings [here](https://github.com/KratosMultiphysics/Kratos/blob/master/.github/workflows/configure.cmd#L40) such that they will be turned into errors in the CI? In this PR

I can set warnings level to /WX (treat warnings as errors), though I think this will 

> in order to keep the compilation clean in the future, could you please add the warnings [here](https://github.com/KratosMultiphysics/Kratos/blob/master/.github/workflows/configure.cmd#L40) such that they will be turned into errors in the CI? In this PR

ok, lets give it a try

**Comment by ddiezrod** [comment]
> in order to keep the compilation clean in the future, could you please add the warnings [here](https://github.com/KratosMultiphysics/Kratos/blob/master/.github/workflows/configure.cmd#L40) such that they will be turned into errors in the CI? In this PR

ok, lets give it a try

**Comment by rubenzorrilla** [comment]
It seems that the`MASS_SOURCE` is not used by the DEM (there are no changes neither in the `DEMApplication` nor in the `SwimmingDEMApplication` aside of those ones required to remove the variable). If this is the case I'd directly remove them from the `dem_variables.h` and keep it in the `FluidDynamicsApplication` variables.

File: .github/workflows/configure.cmd [hunk 2]
```diff
@@ -37,7 +37,7 @@ cmake                                                ^
   -B"%KRATOS_BUILD%\%KRATOS_BUILD_TYPE%"             ^
   -DBOOST_ROOT="%BOOST_ROOT_1_72_0%"                 ^
   -DINSTALL_RUNKRATOS=OFF                            ^
-  -DCMAKE_CXX_FLAGS="/Od /we4661 /we4804"            ^
+  -DCMAKE_CXX_FLAGS="/Od /we4661 /we4804 /WX"        ^
```
Comment (philbucher) [inline .github/workflows/configure.cmd:40]: lets see if we manage this, would be great IMO!

**Comment by ddiezrod** [comment]
It seems one of the warnings comes from dem application.

https://github.com/KratosMultiphysics/Kratos/blob/9735c7a197bba46058f23174038115258b1fc2d2/applications/DEMApplication/custom_strategies/strategies/explicit_solver_continuum.cpp#L585

Visual Studio is not able to recognize the bold print option (\e[m1 e[m0). Would it be ok to just use the regular print? @maceligueta

**Comment by ddiezrod** [comment]
Any idea in the warning in structural mechanics? It seems it complains about the deprecated constuctor, but its not actually being used I think...

**Comment by philbucher** [comment]
@maceligueta @GuillermoCasas the SwimmingDEM app still uses the old interfaces => #7381 
We want to remove this by the end of this month, can you please update it?

**Comment by maceligueta** [comment]
> @maceligueta @GuillermoCasas the SwimmingDEM app still uses the old interfaces => #7381
> We want to remove this by the end of this month, can you please update it
Sure @philbucher , I will dedicate some time to it right today

File: applications/DEMApplication/custom_strategies/strategies/explicit_solver_continuum.cpp [hunk 3]
```diff
@@ -582,7 +582,7 @@ namespace Kratos {
             }//while
 
                 if (iteration < maxiteration){
-                KRATOS_INFO("DEM") << "The iterative procedure converged after " << iteration << " iterations, to value \e[1m" << current_coordination_number << "\e[0m using a global amplification of radius of " << amplification << ". " << "\n" << std::endl;
+                KRATOS_INFO("DEM") << "The iterative procedure converged after " << iteration << " iterations, to value " << current_coordination_number << " using a global amplification of radius of " << amplification << ". " << "\n" << std::endl;
```
Comment (philbucher) [inline applications/DEMApplication/custom_strategies/strategies/explicit_solver_continuum.cpp:585]: @maceligueta is this ok?
If you insist on keeping it then I think the easiest would be to use the macros to check where Kratos is compiled

Comment (philbucher) [reply applications/DEMApplication/custom_strategies/strategies/explicit_solver_continuum.cpp:585]: @maceligueta I just saw that you gave your thumbs up already, guess then it is ok

**Comment by philbucher** [comment]
Now it compiles!!!
IMO only very minor changes were necessary, for me this ready to go in
@RiccardoRossi @roigcarlo are you ok with the changes regarding the global pointer related things?

**Comment by rubenzorrilla** [comment]
No more concerns from the fluid side. Thanks!!!

File: applications/DEMApplication/custom_strategies/strategies/explicit_solver_continuum.cpp (hunk 3 above)
Comment (maceligueta) [reply applications/DEMApplication/custom_strategies/strategies/explicit_solver_continuum.cpp:585]: Actually, I removed this 'font formatting' from a couple of places more (long ago) and this one remained unintentionally.

**Comment by roigcarlo** [comment]
Also good from our side! gj

**Review by philbucher — APPROVED** [review]
:tada:

**Comment by ddiezrod** [comment]
Thanks guys, I think this is a quite positive change, lets hope we can maintain it.

**Comment by philbucher** [comment]
> lets hope we can maintain it.

well now the CI enforces it :D :tada:

**Pull request merged**